### PR TITLE
implemented #2978 mysqlutil: using auto_increment can lead into inval…

### PR DESCRIPTION
**Pull request by pvanek**
…id DDL in some cases

### CODE REVIEW

File: examples/test/qlib/SqlUtil/MysqlSqlUtil.qtest [hunk 1]
```diff
@@ -178,6 +223,23 @@ class MysqlTest inherits SqlTestBase {
...
 
+    private testCreateTableWithAutoIncrementNegative() {
+        MysqlTestAutoIncrementNegativeSchema schema(table.getDatasource());
+        try {
+            schema.align(False, m_options.verbose);
+        }
+        catch (hash ex) {
+            assertEq("MYSQL-COLUMN-ERROR", ex.err, "exception check");
+            assertEq(True, ex.desc =~ /AUTO_INCREMENT can be used only with a subset of types/, "exception check");
+        }
```
Comment (sejvlond) [inline examples/test/qlib/SqlUtil/MysqlSqlUtil.qtest]: if the exception wouldn't be thrown, the test will also pass. use 
```
assertThrows("MYSQL-COLUMN-ERROR", 
              "AUTO_INCREMENT can be used only with a subset of types",
              sub() { schema.align(False, m_options.verbose); });
```

**Review by sejvlond — CHANGES_REQUESTED** [review]

File: examples/test/qlib/SqlUtil/MysqlSqlUtil.qtest (hunk 1 above)
Comment (pvanek) [reply examples/test/qlib/SqlUtil/MysqlSqlUtil.qtest]: done, sir!

Comment (sejvlond) [reply examples/test/qlib/SqlUtil/MysqlSqlUtil.qtest]: sir, there is the "resolve conversation" button, sir

File: examples/test/qlib/SqlUtil/MysqlSqlUtil.qtest [hunk 2]
```diff
@@ -79,6 +79,48 @@ class MysqlTestSchema inherits SqlUtilTestSchema {
...
+    }
+}
+
+class MysqlTestAutoIncrementPositiveSchema inherits AbstractSchema {
+
+    constructor(AbstractDatasource ds, *string dts, *string its)
+        : AbstractSchema(ds, dts, its)
+    {
+    }
+
```
Comment (sejvlond) [inline examples/test/qlib/SqlUtil/MysqlSqlUtil.qtest:108]: there is an empty line between constructor and methods, but in `MysqlTestAutoIncrementNegativeSchema` there is none, but you know what? I will forgive you that! Next time, pay more attention! :dagger:  :smile_cat:

**Review by sejvlond — APPROVED** [review]

**Pull request merged**